### PR TITLE
Revert JN messages to the original ones

### PR DESCRIPTION
**Pull request by Fi3**

### CODE REVIEW

File: protocols/v2/binary-sv2/serde-sv2/src/error.rs [hunk 1]
```diff
@@ -106,144 +111,3 @@ impl Display for Error {
         }
     }
 }
-
-// impl core::error::Error for Error {}
```
Comment (darricksee) [inline protocols/v2/binary-sv2/serde-sv2/src/error.rs:110]: Why remove all these tests?

File: roles/v2/pool/src/lib/job_negotiator/mod.rs [hunk 2]
```diff
@@ -32,26 +49,76 @@ impl JobNegotiatorDownstream {
...
     }
-    pub fn stay_alive(self_mutex: Arc<Mutex<Self>>) {
+    pub fn start(self_mutex: Arc<Mutex<Self>>) {
+        let recv = self_mutex.safe_lock(|s| s.receiver.clone()).unwrap();
         tokio::spawn(async move {
             loop {
-                let recv = self_mutex.safe_lock(|s| s.receiver.clone()).unwrap();
-                let _ = recv.recv().await;
+                if let Ok(message) = recv.recv().await {
+                    let mut frame: StdFrame = message.try_into().unwrap();
```
Comment (coleFD) [inline roles/v2/pool/src/lib/job_negotiator/mod.rs:57]: handle unwraps by sending to the main thread. if a bad message is sent this task will fail while the others will keep running

File: roles/v2/pool/src/lib/job_negotiator/mod.rs [hunk 3]
```diff
@@ -32,26 +49,76 @@ impl JobNegotiatorDownstream {
...
+impl ParseClientJobNegotiationMessages for JobNegotiatorDownstream {
+    fn handle_allocate_mining_job(
+        &mut self,
+        message: AllocateMiningJobToken,
+    ) -> Result<roles_logic_sv2::handlers::job_negotiation::SendTo, roles_logic_sv2::Error> {
+        let res = JobNegotiation::AllocateMiningJobTokenSuccess(AllocateMiningJobTokenSuccess {
+            request_id: message.request_id,
+            mining_job_token: get_random_token(),
+            coinbase_output_max_additional_size: crate::COINBASE_ADD_SZIE,
+            coinbase_output: self.coinbase_output.clone().try_into().unwrap(),
```
Comment (coleFD) [inline roles/v2/pool/src/lib/job_negotiator/mod.rs:89]: can you comment why the unwrap is safe

**Review by coleFD — CHANGES_REQUESTED** [review]

File: protocols/v2/binary-sv2/serde-sv2/src/error.rs (hunk 1 above)
Comment (Fi3) [reply protocols/v2/binary-sv2/serde-sv2/src/error.rs:110]: not sure how useful the tests are. I think that removing them improve the codebase (reduce size) and do not worsen it in any significant way. Btw if you prefer I can re-add them.

File: protocols/v2/binary-sv2/serde-sv2/src/primitives/short_tx_id.rs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        Ok(ShortTxId(Inner::Ref(value)))
+    }
+
+    fn visit_byte_buf<E>(self, v: Vec<u8>) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        if v.len() >= 6 {
+            let v = v[..6].to_vec();
+            Ok(ShortTxId(Inner::Owned(Box::new(v.try_into().unwrap()))))
```
[the remark this answers is not in the document]
Comment (Fi3) [reply protocols/v2/binary-sv2/serde-sv2/src/primitives/short_tx_id.rs:98]: yes they are views in bigger arrays

File: protocols/v2/subprotocols/job-negotiation/src/allocate_mining_job_token.rs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#[cfg(not(feature = "with_serde"))]
+use binary_sv2::binary_codec_sv2;
+use binary_sv2::{Deserialize, Serialize, Str0255, B0255, B064K};
+use core::convert::TryInto;
+
+/// ## AllocateMiningJobToken (Client -> Server)
+/// A request to get an identifier for a future-submitted mining job.
+/// Rate limited to a rather slow rate and only available on connections where this has been
+/// negotiated. Otherwise, only `mining_job_token(s)` from `CreateMiningJob.Success` are valid.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
```
[the remark this answers is not in the document]
Comment (Fi3) [reply protocols/v2/subprotocols/job-negotiation/src/allocate_mining_job_token.rs:12]: How can I do that?

File: roles/translator/src/job_negotiator/mod.rs [hunk 6]
```diff
@@ -32,19 +34,17 @@ use crate::proxy_config::ProxyConfig;
...
 impl JobNegotiator {
     pub async fn new(
         address: SocketAddr,
         authority_public_key: [u8; 32],
-        sender_coinbase_output_max_additional_size: Sender<(CoinbaseOutputDataSize, u64)>,
-        sender_coinbase_out: Sender<(Vec<TxOut>, u64)>,
+        sender_coinbase_output_max_additional_size: Sender<(CoinbaseOutputDataSize, Vec<u8>)>,
+        sender_coinbase_out: Sender<(Vec<TxOut>, Vec<u8>)>,
         config: ProxyConfig,
     ) -> Arc<Mutex<Self>> {
```
[the remark this answers is not in the document]
Comment (Fi3) [reply roles/translator/src/job_negotiator/mod.rs:49]: allocate token can not panic

File: roles/translator/src/job_negotiator/mod.rs [hunk 7]
```diff
@@ -102,28 +101,36 @@ impl JobNegotiator {
...
+                        .unwrap()];
+                        sender_out_script.send((out_script, token)).await.unwrap();
                     }
                     Ok(_) => unreachable!(),
                     Err(_) => todo!(),
                 }
             }
         });
     }
+    async fn allocate_token(self_mutex: Arc<Mutex<Self>>) {
```
[the remark this answers is not in the document]
Comment (Fi3) [reply roles/translator/src/job_negotiator/mod.rs:127]: can not panic

**Pull request merged**